### PR TITLE
Create beacon icon

### PR DESCRIPTION
**Pull request by Incompleteusern**
Also fixes cursor issue

image

### CODE REVIEW

File: common/src/main/java/com/wynntils/features/user/overlays/map/MinimapFeature.java [hunk 1]
```diff
@@ -205,47 +247,81 @@ public void render(PoseStack poseStack, float partialTicks, Window window) {
...
+
+            float toSquareScale = 1f;
+
+            if (maskType == MapMaskType.Rectangular) {
+                // Scale as necessary
+                toSquareScale =
+                        Math.min(scaledWidth / Math.abs(compassOffsetX), scaledHeight / Math.abs(compassOffsetZ)) / 2;
+            } else if (maskType == MapMaskType.Circle) {
+                toSquareScale = scaledWidth
+                        / (MathUtils.magnitude(compassOffsetX, compassOffsetZ * scaledWidth / scaledHeight))
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/user/overlays/map/MinimapFeature.java:282]: feels a bit weird to call it like this here

File: common/src/main/java/com/wynntils/features/user/overlays/map/MinimapFeature.java [hunk 2]
```diff
@@ -205,47 +247,81 @@ public void render(PoseStack poseStack, float partialTicks, Window window) {
...
+
+            float compassOffsetX = compassRenderX - centerX;
+            float compassOffsetZ = compassRenderZ - centerZ;
+            float distance = MathUtils.magnitude(compassOffsetX, compassOffsetZ);
+
+            compassOffsetX /= distance;
+            compassOffsetZ /= distance;
+
+            float scaledWidth = width - 4 * compassSize;
+            float scaledHeight = height - 4 * compassSize;
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/user/overlays/map/MinimapFeature.java:272]: why -4? seems random

File: common/src/main/java/com/wynntils/features/user/overlays/map/MinimapFeature.java [hunk 3]
```diff
@@ -205,47 +247,81 @@ public void render(PoseStack poseStack, float partialTicks, Window window) {
...
 
-            if (!followPlayerRotation) {
-                poseStack.popPose();
+            Pair<Float, Float> compassLocation = getRenderLocation(
+                    compass, playerX, playerZ, centerX, centerZ, sinRotationRadians, cosRotationRadians);
+            float compassRenderX = compassLocation.a();
+            float compassRenderZ = compassLocation.b();
+
+            final float compassSize = Math.max(compass.getWidth(), compass.getHeight()) * 0.8f * poiScale;
+
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/user/overlays/map/MinimapFeature.java:263]: 0.8 is just scaling down right? Is the texture too big?

File: common/src/main/java/com/wynntils/wynn/model/map/poi/WaypointPoi.java [hunk 4]
```diff
@@ -7,8 +7,16 @@
 import com.wynntils.gui.render.Texture;
 
 public class WaypointPoi extends IconPoi {
+
+    private final PointerPoi pointer;
+
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/wynn/model/map/poi/WaypointPoi.java:12]: feels a bit weird on first look but also clever in a way. should be fine to keep this

File: common/src/main/java/com/wynntils/features/user/overlays/map/MinimapFeature.java (hunk 2 above)
Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/features/user/overlays/map/MinimapFeature.java:272]: It is random

File: common/src/main/java/com/wynntils/features/user/overlays/map/MinimapFeature.java (hunk 3 above)
Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/features/user/overlays/map/MinimapFeature.java:263]: Copied from legacy

File: common/src/main/java/com/wynntils/wynn/model/map/poi/WaypointPoi.java (hunk 4 above)
Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/wynn/model/map/poi/WaypointPoi.java:12]: I don't see a better alternative for now

File: common/src/main/java/com/wynntils/features/user/overlays/map/MinimapFeature.java (hunk 1 above)
Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/features/user/overlays/map/MinimapFeature.java:282]: Call what?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/overlays/map/MinimapFeature.java:282]: toSquareScale when map is circle?

Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/features/user/overlays/map/MinimapFeature.java:282]: Oops

**Review by kristofbolyai — APPROVED** [review]

**Pull request merged**